pkg/controllers/rollout/controller.go

```diff
@@ -119,10 +119,12 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	}
 	klog.V(2).InfoS("Picked the bindings to be updated", "clusterResourcePlacement", crpName, "numberOfBindings", len(toBeUpdatedBindings))
 
-	// update all the bindings in parallel according to the rollout plan, requeue the request after the mean value of unavailable period
-	// has passed so that some applied bindings will be considered "ready" by the rollout process regardless if the binding updates succeed or not.
-	// This is to avoid the case that the rollout process stalling because the time based binding readiness does not trigger any event.
-	return ctrl.Result{RequeueAfter: time.Duration(*crpCopy.Spec.Strategy.RollingUpdate.UnavailablePeriodSeconds/2) * time.Second},
+	// Update all the bindings in parallel according to the rollout plan.
+	// We need to requeue the request regardless if the binding updates succeed or not
+	// to avoid the case that the rollout process stalling because the time based binding readiness does not trigger any event.
+	// We wait for 1/5 of the UnavailablePeriodSeconds so we can catch
+	// TODO: only wait the time we need to wait for the first applied but not ready binding to be ready
```

Comment:
if i understand it correctly, you want to use the minimal wait time returned by the  isBindingReady?

if so, in the isBindingReady func, we could use "UnavailablePeriodSeconds" when spec is not applied yet, instead of -1.

Reply:
the isBindingReady doesn't have that info